### PR TITLE
Test updates for common::testing module.

### PR DESCRIPTION
**Pull request by mjcarroll**
Update ignition-common to make use of ign-common::testing

* Builds on top of #314

### CODE REVIEW

File: src/Timer_TEST.cc [hunk 1]
```diff
@@ -41,6 +41,7 @@ TEST(Timer_TEST, Sequence)
 
   t.Stop();
   EXPECT_FALSE(t.Running());
+  std::this_thread::sleep_for(std::chrono::milliseconds(1));
```
Comment (nkoenig) [inline src/Timer_TEST.cc:44]: Is this addition needed?

**Review by nkoenig — APPROVED** [review]
One minor comment

File: src/Timer_TEST.cc (hunk 1 above)
Comment (mjcarroll) [reply src/Timer_TEST.cc]: No, I believe the flakiness was addressed via https://github.com/ignitionrobotics/ign-common/pull/343

Removed.
